modules/oe_theme_helper/src/Plugin/field_group/FieldGroupFormatter/InPageNavigation.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_helper\Plugin\field_group\FieldGroupFormatter;
+
+use Drupal\Core\Template\Attribute;
+use Drupal\Core\Render\Element;
+use Drupal\Component\Utility\Html;
+
+/**
+ * Display a field group using the ECL in-page navigation component.
+ *
+ * @FieldGroupFormatter(
+ *   id = "oe_theme_helper_in_page_navigation",
+ *   label = @Translation("In-page navigation"),
+ *   description = @Translation("Display a field group using the ECL in-page navigation component."),
+ *   supported_contexts = {
+ *     "view"
+ *   }
+ * )
+ */
+class InPageNavigation extends InPageNavigationBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preRender(&$element, $rendering_object) {
+    parent::preRender($element, $rendering_object);
+
+    $element += [
+      '#type' => 'oe_theme_helper_in_page_navigation',
+      '#title' => $this->getLabel(),
+    ];
+
+    // Build page contents based on titles from the In-page navigation item
+    // groups.
+    $links = [];
+    $children = Element::children($element);
+    $length = count($children);
+    $i = 0;
+    foreach ($children as $value) {
+      // Mark first and last elements. It is helpful in theming.
+      if ($i === 0) {
+        $element[$value]['#first'] = TRUE;
+      }
+      if ($i === $length - 1) {
+        $element[$value]['#last'] = TRUE;
```

Comment:
I use it in modules/oe_theme_helper/templates/oe-theme-helper-in-page-navigation-item.html.twig.